### DISCUSSION TITLE
The Service Controls on Mainsail aren't working for me.

### DISCUSSION THREAD

**Discussion by Heiko910** [bassamanator/Sovol-SV06-firmware#138] [General]
The Service Controls and Host Controls of Mainsail (upper Left corner of the page) don't seem to work. If I I click on Any of the Service Controls or Host Controls nothing seems to happen.
![Klipper Controls](https://github.com/bassamanator/Sovol-SV06-firmware/assets/145156088/e4feafaa-b336-4be2-b42b-7285b8c2f3ca)

**To Reproduce**
Just open Mainsail and try to reboot the Host or restart KlipperScreen. Does that work for you?

**Expected behavior**
I would expect that clicking on one of the Service Controls it would either stop or restart the service, but that is not the case.

**What printer are you using?**
Sovol Sv06

**What kind of device are you using to host Klipper?**

Inovato Quadra Running Armbian

**How did you install Klipper?**

KIAUH

**Is your local copy of this configuration up to date?**
All services and also the OS are up to date.

**What slicer are you using?**

- [ x] PrusaSlicer
- [ ] SuperSlicer
- [ x] Cura
- [ x] OrcaSlicer

**Are you using the config bundles from this repo?**

- [ x] Yes
- [ ] No

**Did you [Adjust your slicer](https://github.com/bassamanator/Sovol-SV06-firmware#adjust-your-slicer)**?

- [ x] Yes
- [ ] No

**Additional context**
Not really a slicer issue but rather a Klipper issue I am assuming.
Please help. I would love to be able to restart services or the host from within Mainsail.
Thanks

**Comment by bassamanator** [comment]
Your user doesn't have appropriate privileges to initiate those actions. Fixable if the user is added to the appropriate group/s. Search for those terms.

**Reply by Heiko910** [reply]
Do you have more specifics. I did some googling but I am coming up empty handed.
Thanks

**Comment by bassamanator** [comment]
Paste the output from `groups $USER`.

**Reply by Heiko910** [reply]
Here is the output:

quadra@inovato:~$ groups $USER
quadra : quadra tty disk dialout sudo audio video plugdev games users systemd-jo
urnal input netdev ssh bluetooth

**Comment by bassamanator** [comment]
ssh into the machine, then type that command.

**Reply by Heiko910** [reply]
Here is the output:

quadra@inovato:~$ groups $USER
quadra : quadra tty disk dialout sudo audio video plugdev games users systemd-jo
urnal input netdev ssh bluetooth

**Comment by bassamanator** [comment]
Here's my output. 
```
pi : pi adm tty dialout cdrom sudo audio video plugdev games users input render netdev spi i2c gpio
```
You can adjust your user's groups via `sudo usermod -aG groupName $USER`.

This might not be related to the groups at all actually. I would suggest you ask at mainsail/klipper/voron discord server.

**Reply by Heiko910** [reply]
I already posted on the mainsail discord but so far no joy.
Thanks for your help

**Comment by Heiko910** [comment]
Hi Bassamanator,
I thought I just share with the fix for the issue that the Service Controls and the Host Controls on the Mainsail Interface wouldn't work for me. 
After a short discussion on the Mainsail Discord I managed to find the solutions.

https://docs.mainsail.xyz/faq/moonraker_warnings/policykit-permissions

This Documentation outlines the install of a Policykit for Moonraker and as soon as I installed that all the restart and Shutdown functions in Mainsail worked as expected.
Thanks again for you help.
Heiko

**Comment by bassamanator** [comment]
Great! I'll know what to tell the next user who encounters this issue. Thanks.
